src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Subscriptions/DependencyRulesSubscriber.cs

```diff
@@ -186,11 +186,6 @@ private IReadOnlyCollection<string> GetWatchedRules(RuleHandlerType handlerType)
 
             await _tasksService.LoadedProjectAsync(async () =>
             {
-                if (_tasksService.UnloadCancellationToken.IsCancellationRequested)
-                {
-                    return;
```

Comment:
Was this doing anything useful? Doesn't `LoadedProjectAsync` prevent unloading during the callback anyway?

Reply:
This line is effectively what LoadProjectAsync does before calling this callback, it won't start the callback if this is true.